tests/test_basic_operation.py

```diff
@@ -89,33 +89,40 @@ def test_013_drop_secondary():
     node1.run()
     assert node1.wait_until_state(target_state="secondary")
     node1.drop()
+    assert not node1.pg_is_running()
     assert node2.wait_until_state(target_state="single")
+    
 
 def test_014_add_new_secondary():
     global node3
-    node3 = cluster.create_datanode("/tmp/node3")
+    node3 = cluster.create_datanode("/tmp/basic/node3")
     node3.create()
     node3.run()
     assert node3.wait_until_state(target_state="secondary")
     assert node2.wait_until_state(target_state="primary")
 
-def test_015_multiple_manual_failover_verify_replication_slot_removed():
-    monitor.failover()
-    assert node3.wait_until_state(target_state="primary")
-    assert node2.wait_until_state(target_state="secondary")
-    node2_replication_slots = node2.run_sql_query("select count(*) from pg_replication_slots")
-    assert node2_replication_slots == [(0,)]
-    node3_replication_slots = node3.run_sql_query("select count(*) from pg_replication_slots")
-    assert node3_replication_slots == [(1,)]
 
-    monitor.failover()
-    assert node2.wait_until_state(target_state="primary")
-    assert node3.wait_until_state(target_state="secondary")
-    node2_replication_slots = node2.run_sql_query("select count(*) from pg_replication_slots");
-    assert node2_replication_slots == [(1,)]
-    node3_replication_slots = node3.run_sql_query("select count(*) from pg_replication_slots");
-    assert node3_replication_slots == [(0,)]
+def test_015_multiple_manual_failover_verify_replication_slot_removed():
+   monitor.failover()
+   assert node3.wait_until_state(target_state="primary")
+   assert node2.wait_until_state(target_state="secondary")
+   node2_replication_slots = node2.run_sql_query("select count(*) from pg_replication_slots")
+   assert node2_replication_slots == [(0,)]
+   node3_replication_slots = node3.run_sql_query("select count(*) from pg_replication_slots")
+   assert node3_replication_slots == [(1,)]
+   
+   node2.fail()
+   node2.run()
```

Comment:
That's the test workaround we have been talking about. It would be best that we don't need it of course.